mqt/bench/utils/utils.py

```diff
@@ -116,10 +116,13 @@ def __init__(self, probability):
         self._theta_p = 2 * np.arcsin(np.sqrt(probability))
         self.ry(2 * self._theta_p, 0)
 
-    def power(self, k):
+    def __eq__(self, other):
```

Comment:
Why has this method been added, if it is not used?

Reply:
PyCharm/CodeQL was complaining.
It seems to be recommended/best practice to provide an override for the `__eq__` operator whenever a derived class adds a property to the class. Kind of understandable, since you typically want to consider that new property for any equality check.